src/components/ui/Chat/Message/index.tsx

```diff
@@ -0,0 +1,62 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { HTMLAttributes } from 'react';
+
+import { BaseProps } from '../../Base';
+import { StyledMessage, StyledMessageInfo, StyledMessageContent } from './Styled';
+import PopOver from '../../PopOver';
+import PopOverItem, { PopOverItemProps } from '../../PopOver/PopOverItem';
+import { Dots } from '../../icons';
+
+export type MessageVariant = 'outgoing' | 'incoming';
+
+export interface MessageProps extends Omit<HTMLAttributes<HTMLDivElement>, 'css'>, BaseProps {
+    /* determines styling for outgoing and incoming messages. */
+    variant: MessageVariant;
+    /* the name of the user that sent the message. */
+    name: string;
+    /* The timestamp the message was created. */
+    time: string;
+    /* The text content of the message. */
+    content: string;
+    /* The options available in the more options button. */
+    moreOptions?: PopOverItemProps[] | null;
+    /* Adds the bubble caret to a message. */
+    showCaret?: boolean;
+}
+
+const Message = (props: MessageProps) => {
```

Comment:
I know this is out of scope for the initial release, but just in terms of naming - you might want to think about Markdown and attachments and other message variants. For example, if you plan on just creating separate components for a MarkdownMessage and AttachmentMessage, should this be a TextMessage? Or do you think you'll end up managing that all through props with a single component?

Reply:
I haven't gotten that far ahead. Initially, I believe we can handle an attachment inside this same component as a prop. If you have some vision around markdown and would recommend naming this to be a different component, I'm open to that.

Reply:
Nope no vision, it was just a general comment to ask if we were thinking about that stuff. Sounds like we are though!